src/components/main-page/main-page.tsx

```diff
@@ -1,11 +1,19 @@
 import * as React from 'react';
-import { HeadFC, PageProps } from 'gatsby';
-import { LecturesGrid, LogoLink } from '~/components';
-import { Footer, Header, SEO } from '~/partials';
+import { LecturesGrid, LogoLink } from './components/components';
+import { Footer, Header } from '~/partials';
+import { MainPageContext } from '~/types';
 
 import '~/assets/stylesheets/index.css';
 
-const IndexPage: React.FC<PageProps> = () => {
+type Props = {
+  pageContext: MainPageContext;
+};
+
+const MainPage: React.FC<Props> = ({
+  pageContext: {
+    allMdx: { edges },
+  },
+}) => {
   return (
```

Comment:
Don't pass edges to this page, extract data from edges in createPage level and pass it, type for context will be like:
```typescript
type MainPageContext = {
  lectures: {}[]
}
```